flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -857,28 +857,30 @@ object AggregateUtil {
       aggregateCalls: Seq[AggregateCall],
       inputType: RelDataType,
       needRetraction: Boolean)
-  : (Array[Int], Array[TableAggregateFunction[_ <: Any]]) = {
+  : (Array[util.List[Integer]], Array[TableAggregateFunction[_ <: Any]]) = {
```

Comment:
Can we use an `Array[Int]` instead of a `List[Integer]`?

Scala `Int` is compiled to Java `int` primitive. `java.lang.Integer` is a boxed type.
Also an array access is faster than a `List.get()` access (`ArrayList.get()` would be a bit better).